### DISCUSSION TITLE
Impact of file-log on production environment

### DISCUSSION THREAD

**Discussion by ponsakorn30214** [Kong/kong#8305] [Help] [answered]
Hi,

According to Kong plugin documentation about file-log plugin
> This plugin uses blocking I/O, which could affect performance when writing to physical files on slow (spinning) disks.

ref: https://docs.konghq.com/hub/kong-inc/file-log/

What would be the impact of using this plugin on the production environment?
I'm thinking of Kong with an average of 60,000 requests per day on CentOS machine with an HDD.
Should I still use file-log plugin in this instance?

**Comment by bungle** [comment, marked as answer]
non-blocking here means that while file is being written, the worker cannot do anything else, e.g. serve other requests. The network io, in OpenResty is non-blocking, which means the whenever the network is waiting, e.g. connection to be established, or data becoming available, the workers can serve other requests. As there is no non-blocking file io generally available (perhaps through mmap), it may affect the performance. But your filesystem can always be memory based or just on some really fast NVMe disks. You need to measure what effect it would have. It might not be something to worry about, but the warning is there so that people think about it.
